### PR TITLE
feat: log permission access

### LINKED ISSUE #2514: Default to always printing log message when permissions are accessed
So users would get by default a list of files accessed even if --allow-read is on.

Use an emoji of some sort to distinguish them from normal output. Maybe something like
```
⚠️  Read granted "/etc/passwd"
```

Should log to stderr.

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes #2514 

Example output:
```
$ ./target/debug/deno --allow-net --allow-read https://deno.land/std/http/file_server.ts
ℹ️  Granted network access to "0.0.0.0:4500"
HTTP server listening on http://0.0.0.0:4500/
ℹ️  Granted read access to "/Users/biwanczuk/dev/deno"
ℹ️  Granted read access to "/Users/biwanczuk/dev/deno"

$ ./target/debug/deno --allow-net --allow-read --no-prompt https://deno.land/std/http/file_server.ts
HTTP server listening on http://0.0.0.0:4500/
```
Screenshot for reference:
Zrzut ekranu 2019-06-14 o 16 13 19

I'm open to @kitsonk's suggestion to introduce another level of internal logging: now we have `-D/--log-debug`, how about we introduce `-I/--log-info`?

### CODE REVIEW

**Comment by hayd** [comment]
> how about we introduce -I/--log-info?

:+1:

**Comment by kitsonk** [comment]
Or how about we introduce `--log-level=debug` or something that means we are not proliferating a load of flags.

**Comment by bartlomieju** [comment]
> Or how about we introduce `--log-level=debug` or something that means we are not proliferating a load of flags.

Done! Added `--log-level/-L` flag that takes either `info` or `debug` value. 
(I don't really like `-L` 🙄... let you be the judge) 

I kept `-D/--log-debug` flag for compatibility - basically it's an alias for `--log-level=debug`.

Now permission access show only with `--log-level=info` or `--log-level=debug`:
```sh
deno_dev --log-level=info -A https://deno.land/std/http/file_server.ts
ℹ️  Granted network access to "0.0.0.0:4500"
HTTP server listening on http://0.0.0.0:4500/
ℹ️  Granted read access to "/Users/biwanczuk/dev/deno"
ℹ️  Granted read access to "/Users/biwanczuk/dev/deno"
ℹ️  Granted read access to "/Users/biwanczuk/dev/deno/Cargo.toml"
ℹ️  Granted read access to "/Users/biwanczuk/dev/deno/buildtools"
ℹ️  Granted read access to "/Users/biwanczuk/dev/deno/tools"
...
```

**Comment by ry** [comment]
The naming of these type of flags has been done before. We'd probably be best off by using existing nomenclature.

In glog ( https://godoc.org/github.com/golang/glog ) they use `-v=$LEVEL` ... I'm most familiar with glog flags like `-logtostderr`... But I know many people hate glog. Does anyone have alternative logging systems we could steal ideas from?

**Comment by kitsonk** [comment]
Previous art:

* Rust is an environment variable `RUST_LOG` so you get `RUST_LOG=info`
* Python is `--log` so `--log=INFO`
* npm uses `--loglevel` so you get `--loglevel verbose` but also has individual flags to control things as well.

There were a few other examples, but the best thing is that they are totally inconsistent.  Some have just simple flags, some only have environment variables.  🤷‍♂ It does seem like `DENO_LOG` could/should be respected though.

**Comment by bartlomieju** [comment]
> In glog ( https://godoc.org/github.com/golang/glog ) they use `-v=$LEVEL`

I like this approach - `-v/--verbosity` seems very intuitive. 

By default we log out only error messages, setting `-v=DEBUG` would be equivalent of current `-D`, while `-v=INFO` would show permission prompts.

I also agree with @kitsonk on handling `DENO_LOG` env variable as well.

File: cli/flags.rs [hunk 1]
```diff
@@ -127,10 +128,19 @@ To get help on the another subcommands (run in this case):
...
     .arg(
+      Arg::with_name("log-level")
+        .short("L")
+        .long("log-level")
+        .help("Set log level")
+        .takes_value(true)
+        .possible_values(&["debug", "info"])
+        .global(true),
+    ).arg(
       Arg::with_name("log-debug")
```
Comment (ry) [inline cli/flags.rs]: Can we just remove this one now?

File: cli/permissions.rs [hunk 2]
```diff
@@ -328,6 +359,17 @@ impl DenoPermissions {
     permission_prompt(message)
   }
 
+  fn log_perm_access(&self, message: &str) {
+    if log_enabled!(log::Level::Info) {
+      eprintln!(
+        "{}",
+        Style::new()
+          .bold()
+          .paint(format!("ℹ️  Granted {}", message))
```
Comment (ry) [inline cli/permissions.rs]: I think we should keep the same emoji for all permission access: ⚠️

Maybe that can be defined as a const at the top of this file?

File: cli/flags.rs (hunk 1 above)
Comment (bartlomieju) [reply cli/flags.rs]: We can, but then for debug you have to go with `-L=debug`, are you fine with that?

Comment (ry) [reply cli/flags.rs]: Yes, I prefer that to having an extra flag

**Pull request merged**